core/src/mender-api.c

```diff
@@ -289,8 +299,8 @@ api_check_for_deployment_v2(int *status, void *response) {
     }
 #endif /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
 #endif /* CONFIG_MENDER_PROVIDES_DEPENDS */
-    /* TODO: Retrieve artifact name from store (see ticket MEN-7479) */
-    if (NULL == cJSON_AddStringToObject(json_provides, "artifact_name", mender_api_config.artifact_name)) {
+
+    if (NULL == cJSON_AddStringToObject(json_provides, "artifact_name", artifact_name)) {
```

Comment:
Relates to my previous comment. The `artifact_name` is expected to be part of the provides already and should not be added here.